app/src/main/java/org/apache/taverna/mobile/ui/tutorial/TutorialActivity.java

```diff
@@ -0,0 +1,200 @@
+        if (!preferencesHelper.isFirstTimeLaunch()) {
+            launchFlashScreen();
+            finish();
+        }
+
+        setContentView(R.layout.activity_tutorial);
+
+        ButterKnife.bind(this);
+
+
+        if (Build.VERSION.SDK_INT >= 21) {
+            getWindow().getDecorView()
+                    .setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+                            | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
+        }
+
+        layouts = new int[]{
+                R.layout.tutorial_slide1,
+                R.layout.tutorial_slide2,
+                R.layout.tutorial_slide3,
+                R.layout.tutorial_slide4};
+
+        addBottomDots(0);
+
+        TutorialSliderAdapter tutorialSliderAdapter = new TutorialSliderAdapter();
+        slidePager.setAdapter(tutorialSliderAdapter);
+        slidePager.addOnPageChangeListener(this);
+
+    }
+
+    @OnClick(R.id.btn_skip)
+    public void skipClick(View v) {
+        launchFlashScreen();
+    }
+
+    @OnClick(R.id.btn_next)
+    public void nextClick(View v) {
+        int current = getItem(+1);
+        if (current < layouts.length) {
+            slidePager.setCurrentItem(current);
+        } else {
+            launchFlashScreen();
+        }
+    }
+
+    public void addBottomDots(int currentPage) {
+        TextView[] dots = new TextView[layouts.length];
+
+        int[] colorsActive = getResources().getIntArray(R.array.array_dot_active);
+        int[] colorsInactive = getResources().getIntArray(R.array.array_dot_inactive);
+
+        dotsLayout.removeAllViews();
+        for (int i = 0; i < dots.length; i++) {
+            dots[i] = new TextView(this);
+            dots[i].setText(Html.fromHtml("&#8226;"));
+            dots[i].setTextSize(35);
+            dots[i].setTextColor(colorsInactive[currentPage]);
+            dotsLayout.addView(dots[i]);
+        }
+
+        if (dots.length > 0)
+            dots[currentPage].setTextColor(colorsActive[currentPage]);
+    }
+
+    public int getItem(int i) {
+        return slidePager.getCurrentItem() + i;
+    }
+
+    public void launchFlashScreen() {
+        preferencesHelper.setFirstTimeLaunch(false);
+        startActivity(new Intent(TutorialActivity.this, FlashScreenActivity.class));
+        finish();
+    }
+
+    @Override
+    public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
+
+    }
+
+    @Override
+    public void onPageSelected(int position) {
+        addBottomDots(position);
+
+        if (position == layouts.length - 1) {
+            bNext.setText(getString(R.string.start));
+            bSkip.setVisibility(View.GONE);
+        } else {
+            bNext.setText(getString(R.string.next));
+            bSkip.setVisibility(View.VISIBLE);
+        }
+
+    }
+
+    @Override
+    public void onPageScrollStateChanged(int state) {
+
+    }
+
+    class TutorialSliderAdapter extends PagerAdapter {
```

Comment:
You have not moved this to the separate class

Reply:
https://www.bignerdranch.com/blog/viewpager-without-fragments/

This link might help.

Reply:
@sagar15795 This helped me alot. Thanks I updated this. 👍